### PR TITLE
Enforce Jenkins Agent Port Settings.

### PR DESCRIPTION
**Pull request by kaiwalyajoshi**
This patch re-introduces setting the `JENKINS_SLAVE_AGENT_PORT` value which was removed in #211 

Mesos currently limits the maximum host port to 32000, the maximum allowable value is 32000.
The current value is set to a random port but the user can easily set this as needed.

This patch also sets `-Djenkins.model.Jenkins.slaveAgentPortEnforce=true` so the user cannot change the port through the UI, changes can only be made by restarting the Marathon App. This was done to prevent confusion.

This PR also refreshes the plugins that are bundled with this version of Jenkins.

### CODE REVIEW

**Review by jeschkies — APPROVED** [review]

File: universe/config.json [hunk 1]
```diff
@@ -72,9 +72,9 @@
                     "default": false
                 },
                 "agent-port": {
-                    "description": "The port that Jenkins agents will use to connect to the master. The default port is set to 50000; a value of 0 will result in a random port being chosen.",
+                    "description": "The port that Jenkins agents will use to connect to the master. The default port is set to 30000; a value of 0 will result in a random port being chosen.",
```
Comment (asekretenko) [inline universe/config.json:75]: Maybe s/a random port/an arbitrary available port/ ? 
After all, the choice is not truly random, and nothing makes this choice before the set of ports available on the agent becomes known.

File: scripts/run.sh [hunk 2]
```diff
@@ -1,7 +1,9 @@
 #!/bin/bash
 
 export LD_LIBRARY_PATH=/libmesos-bundle/lib:/libmesos-bundle/lib/mesos:$LD_LIBRARY_PATH
-export JENKINS_SLAVE_AGENT_PORT=$PORT_AGENT
+# Set JENKINS_SLAVE_AGENT_PORT to the value of networking.agent-port from universe/config.json
+# with requirePorts:true set in the marathon runspec to ensure the port is available.
+export JENKINS_SLAVE_AGENT_PORT=$PORT3
```
Comment (asekretenko) [inline scripts/run.sh:6]: Does `java .. -jar ${JENKINS_FOLDER}/jenkins.war` actually use this environment variable?

From my observations and from the code in https://github.com/jenkinsci/docker/blob/7d1ab73338e76b49c856e07a63550d80eeb35bf3/jenkins.sh#L21 it looks like it doesn't; 
shouldn't we pass the port as an option `-Djenkins.model.Jenkins.slaveAgentPort=...`?

File: universe/config.json (hunk 1 above)
Comment (kaiwalyajoshi) [reply universe/config.json]: Sounds reasonable, applying this.

File: scripts/run.sh (hunk 2 above)
Comment (kaiwalyajoshi) [reply scripts/run.sh]: Good catch!

**Review by asekretenko — APPROVED** [review]

**Pull request merged**